mcs/class/System.Private.CoreLib/System/Exception.cs

```diff
@@ -43,6 +43,8 @@ string GetStackTrace (bool needFileInfo)
 		{
 			if (_stackTraceString != null)
 				return _stackTraceString;
+			if (_traceIps == null)
```

Comment:
Why do you think this is needed?

Reply:
It should roughly be the equivalent of https://github.com/dotnet/coreclr/blob/2520798548b0c414f513aaaf708399f8ef5a4f6c/src/System.Private.CoreLib/src/System/Exception.CoreCLR.cs#L184-L187.

When an exception object is created and the `StackTrace` property is accessed it should return `null` until it is thrown. That's a behavior tested in System.Runtime.Tests and also implemented in the non-NetCore code base. `StackTrace.ToString` always returns non-null value, so it needs to be handled here.

Reply:
I don't think CoreCLR version matters for us, we have quite different runtime implementation and _traceIps == null is already handled in stack trace extraction from the exception

Reply:
It never returns `null` from `ToString` though. The original implementation results in an empty string being returned. Since `StackTrace.ToString` is shared with CoreCLR I don't see any other place to handle this.

Reply:
To make myself clear, I am trying to accomplish `new ApplicationException().StackTrace == null`.

Reply:
oh, makes sense. Could you please contribute a test for this?

Reply:
It's already there... I originally spotted it by running `make run-System.Runtime.Tests`.